app/models/manageiq/providers/cloud_manager/provision_workflow.rb

```diff
@@ -90,6 +88,20 @@ def allowed_ci(ci, relats, filtered_ids = nil)
     super(ci, relats, sources, filtered_ids)
   end
 
+  def availability_zone_to_cloud_network(src)
+    if src[:availability_zone]
+      load_ar_obj(src[:availability_zone]).cloud_subnets.each_with_object({}) do |cs, hash|
```

Comment:
@d-m-u - Would there be a need for RBAC'd `cloud_subnets` here and on line 99?

Reply:
@syncrou isn't that what https://github.com/d-m-u/manageiq/blob/5fa2fce427f36c8d1178df7af44aa5257ebdadcd/app/models/manageiq/providers/cloud_manager/provision_workflow.rb#L30 is doing?

Reply:
@d-m-u @syncrou The caller passes in an rbac filtered list of cloud_networks [here](https://github.com/d-m-u/manageiq/blob/5fa2fce427f36c8d1178df7af44aa5257ebdadcd/app/models/manageiq/providers/cloud_manager/provision_workflow.rb#L31) so there is no need for additional checking at this level.  Only the IDs from the initial filtered list are possible return values.

Reply:
@gmcculloug - Ahh - missed the `get_targets_for_source` that built out the targets list.